template/sample/template.gen.go

```diff
@@ -342,106 +311,83 @@ var (
+				if err != nil {
+					return nil, fmt.Errorf("failed to eval BoolPrimitive for instance '%s': %v", instName, err)
+				}
 
-					if err != nil {
-						result = multierror.Append(result, fmt.Errorf("failed to eval DoublePrimitive for instance '%s': %v", name, err))
-						continue
-					}
+				DoublePrimitive, err := mapper.Eval(castedInst.DoublePrimitive, attrs)
 
-					StringPrimitive, err := mapper.Eval(md.StringPrimitive, attrs)
+				if err != nil {
+					return nil, fmt.Errorf("failed to eval DoublePrimitive for instance '%s': %v", instName, err)
+				}
 
-					if err != nil {
-						result = multierror.Append(result, fmt.Errorf("failed to eval StringPrimitive for instance '%s': %v", name, err))
-						continue
-					}
+				StringPrimitive, err := mapper.Eval(castedInst.StringPrimitive, attrs)
 
-					Int64Map, err := template.EvalAll(md.Int64Map, attrs, mapper)
+				if err != nil {
+					return nil, fmt.Errorf("failed to eval StringPrimitive for instance '%s': %v", instName, err)
+				}
 
-					if err != nil {
-						result = multierror.Append(result, fmt.Errorf("failed to eval Int64Map for instance '%s': %v", name, err))
-						continue
-					}
+				Int64Map, err := template.EvalAll(castedInst.Int64Map, attrs, mapper)
 
-					instances = append(instances, &istio_mixer_adapter_sample_report.Instance{
-						Name: name,
+				if err != nil {
+					return nil, fmt.Errorf("failed to eval Int64Map for instance '%s': %v", instName, err)
+				}
 
-						Value: Value,
+				_ = castedInst
 
-						Dimensions: Dimensions,
+				return &istio_mixer_adapter_sample_report.Instance{
+					Name: instName,
 
-						Int64Primitive: Int64Primitive.(int64),
+					Value: Value,
 
-						BoolPrimitive: BoolPrimitive.(bool),
+					Dimensions: Dimensions,
 
-						DoublePrimitive: DoublePrimitive.(float64),
+					Int64Primitive: Int64Primitive.(int64),
 
-						StringPrimitive: StringPrimitive.(string),
+					BoolPrimitive: BoolPrimitive.(bool),
 
-						Int64Map: func(m map[string]interface{}) map[string]int64 {
-							res := make(map[string]int64, len(m))
-							for k, v := range m {
-								res[k] = v.(int64)
-							}
-							return res
-						}(Int64Map),
-					})
-					_ = md
-				}
+					DoublePrimitive: DoublePrimitive.(float64),
 
-				if err := handler.(istio_mixer_adapter_sample_report.Handler).HandleSample(instances); err != nil {
-					result = multierror.Append(result, fmt.Errorf("failed to report all values: %v", err))
-				}
+					StringPrimitive: StringPrimitive.(string),
 
-				err := result.ErrorOrNil()
-				if err != nil {
-					return status.WithError(err)
-				}
+					Int64Map: func(m map[string]interface{}) map[string]int64 {
+						res := make(map[string]int64, len(m))
+						for k, v := range m {
+							res[k] = v.(int64)
+						}
+						return res
+					}(Int64Map),
+				}, nil
+			},
 
-				return status.OK
+			DispatchReport: func(ctx context.Context, insts interface{}, handler adapter.Handler) (adapter.ReportResult, error) {
```

Comment:
@mandarjog you are right 

Hi Mandar
you are right on that example
so two options..  we re expand the array in the dispatch code code per template.. or get back to doing eval + dispatch together... I would prefer former.. because based on the amt of template code and generated code, this split seem to have reduced some amount of total magic (template work + gererated stuff) code. 
Performance is the only thing I am unclear of. But I think iterating over a array or pointers should be super cheap

Also if we decide to combine we need to be clear on how the generated code will do the error from adapter to status mapping, error from eval to status and be sure that the caller does not care.
Thoughts ?